src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/Namespaces.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    using System.Collections.Generic;
+
+    internal static class Namespaces
+    {
+        internal static readonly IReadOnlyList<string> System = new[]
+        {
+            nameof(System),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemCollectionsGeneric = new[]
+        {
+            nameof(System),
+            nameof(global::System.Collections),
+            nameof(global::System.Collections.Generic),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemThreading = new[]
+        {
+            nameof(System),
+            nameof(global::System.Threading),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemDiagnostics = new[]
+        {
+            nameof(System),
+            nameof(global::System.Diagnostics),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemThreadingTasks = new[]
+        {
+            nameof(System),
+            nameof(global::System.Threading),
+            nameof(global::System.Threading.Tasks),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemRuntimeCompilerServices = new[]
+        {
+            nameof(System),
+            nameof(global::System.Runtime),
+            nameof(global::System.Runtime.CompilerServices),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemRuntimeInteropServices = new[]
+        {
+            nameof(System),
+            nameof(global::System.Runtime),
+            nameof(global::System.Runtime.InteropServices),
+        };
+
+        internal static readonly IReadOnlyList<string> SystemWindowsThreading = new[]
+        {
+            nameof(System),
+            nameof(global::System.Windows),
+            "Threading",
+        };
+
+        internal static readonly IReadOnlyList<string> MicrosoftVisualStudioThreading = new[]
+        {
+            "Microsoft",
+            "VisualStudio",
+            "Threading",
+        };
+
+        internal static readonly IReadOnlyList<string> MicrosoftVisualStudioShell = new[]
+        {
+            "Microsoft",
+            "VisualStudio",
+            "Shell",
+        };
+
+        internal static readonly IReadOnlyList<string> MicrosoftVisualStudioShellInterop = new[]
+        {
+            "Microsoft",
+            "VisualStudio",
+            "Shell",
+            "Interop",
```

Comment:
I didn't expect to see anything VisualStudio specific. Taking note here to make sure the need is confirmed...